mutex_yield.c

```diff
@@ -0,0 +1,73 @@
+#include <libkern.h>
+#include <mutex_yield.h>
+#include <sched.h>
+#include <critical_section.h>
+#include <clock.h>
+
+void mtx_yield_init(mtx_yield_t *mtx) {
+  assert(mtx);
+
+  *mtx = 0;
+}
+void mtx_yield_lock(mtx_yield_t *mtx) {
+  assert(mtx);
+
+  while (true) {
+    while (*mtx == 1)
+      sched_yield(true);
+
+    cs_enter();
+    if (*mtx == 0) {
+      *mtx = 1;
+      cs_leave();
+      return;
+    }
+    cs_leave();
+  }
+}
+void mtx_yield_unlock(mtx_yield_t *mtx) {
+  assert(mtx);
+
+  cs_enter();
+  *mtx = 0;
```

Comment:
I have another version of that mutex with atomic operations, but I delivered this one, because I wasn't completely sure what memory order I should use, what fences etc. One battle at a time :)